Cython/Compiler/Optimize.py

```diff
@@ -2913,6 +2913,27 @@ def _handle_simple_method_dict_setdefault(self, node, function, args, is_unbound
             may_return_none=True,
             utility_code=load_c_utility('dict_setdefault'))
 
+    PyDict_Pop_func_type = PyrexTypes.CFuncType(
+        PyrexTypes.py_object_type, [
+            PyrexTypes.CFuncTypeArg("dict", PyrexTypes.py_object_type, None),
+            PyrexTypes.CFuncTypeArg("key", PyrexTypes.py_object_type, None),
+            PyrexTypes.CFuncTypeArg("default", PyrexTypes.py_object_type, None),
+            ])
+
+    def _handle_simple_method_dict_pop(self, node, function, args, is_unbound_method):
+        """Replace dict.pop() by a call to _PyDict_Pop().
+        """
+        if len(args) == 2:
+            args.append(ExprNodes.NullNode(node.pos))
+        elif len(args) != 3:
+            self._error_wrong_arg_count('dict.pop', node, args, "2 or 3")
+            return node
+
+        return self._substitute_method_call(
+            node, function,
+            "_PyDict_Pop", self.PyDict_Pop_func_type,
```

Comment:
This would probably need a fallback implementation for PyPy. I doubt that it implements this non-official function.

Instead of calling it directly, write a short wrapper function that calls it only in CPython (see the `CYTHON_COMPILING_IN_CPYTHON` macro) and falls back to a generic method call in PyPy, e.g. with `PyObject_CallMethodObjArgs()` or so. You can use `PYIDENT("pop")` to get the method name as a Python `str` constant.

Reply:
Implemented here: c3e1118110c55eb50178ee145b7a10129b2b79b9